.github/workflows/generate-summary.yaml

```diff
@@ -409,10 +409,30 @@ jobs:
       - name: Trim issue length # reduce the number of lines in final issue so github always creates issue
         run: |
           head -c 65000 issue.md > trimmed_issue.md
-          if [ $(stat -c%s issue.md) -gt 65000 ]; then printf "\n\`\`\`\nIssue text has been trimmed. Please check logs for the untrimmed issue.\n" >> trimmed_issue.md; fi
+          if [ $(stat -c%s issue.md) -gt 65000 ]; then
+            echo "\n\`\`\`\nIssue text has been trimmed. Please check the following Gist Link for the untrimmed issue.\n" >> trimmed_issue.md
+            FILE_SIZE=$(stat -c%s "issue.md")
+            ONE_MB_IN_BYTES=1048000
+            if [ "$FILE_SIZE" -gt "$ONE_MB_IN_BYTES" ]; then
+              echo "\nThe file was greater than 1 MB and has been trimmed to fit the Gist limit. Please check logs for the untrimmed issue." > trimmed_issue_gist.md
+            fi
+            head -c $ONE_MB_IN_BYTES "issue.md" >> trimmed_issue_gist.md
+            if [ "$FILE_SIZE" -gt "$ONE_MB_IN_BYTES" ]; then
+              echo "\nThe file was greater than 1 MB and has been trimmed to fit the Gist limit. Please check logs for the untrimmed issue." >> trimmed_issue_gist.md
+            fi
+          fi
           run_id=${{ github.run_id }} && echo "Associated run is: https://github.com/patrick-rivos/gcc-postcommit-ci/actions/runs/$run_id" >> trimmed_issue.md
           cat trimmed_issue.md
 
+      - name: Create Issue Gist
+        run: |
+          if [ -e "trimmed_issue_gist.md" ]; then
+            python3 -m pip install pygithub requests pyopenssl --upgrade
+            python3 ./scripts/create_gist.py --token ${{ secrets.GIST_TOKEN }} --input trimmed_issue_gist.md --output issue_gist_url.txt --title ${{ inputs.issue_hash_prefix }}_${{ inputs.gcchash }}_issue.md
```

Comment:
iirc `inputs.issue_hash_prefix` should already include a `_`

Reply:
Currently `issue_hash_prefix` in run-frequent is
```
  summarize:
    if: "!cancelled()" # Generate github issues even when some (or all) targets fail to build
    needs: [init-submodules, creg, cmreg-self-hosted, cmreg-self-hosted-uc]
    permissions:
      issues: write
    uses: ./.github/workflows/generate-summary.yaml
    with:
      gcchash: ${{ needs.init-submodules.outputs.gcchash }}
      issue_num: ${{ inputs.issue_num }}
      issue_hash_prefix: Testsuite Status
      prefix: ''
      gcc_branch: master
 ```
 should it be Testsuite status_ instead?

Reply:
oops confused issue_hash_prefix with prefix. This should be fine